app/shared/design-system/components/primitives/auto-input.tsx

```diff
@@ -0,0 +1,205 @@
+export class AutoInput extends EventTarget {
+	// PRIVATE
+	#selectTarget = null;
+	#activeTarget = null;
+	#activeEvents = ['keydown', 'focusin', 'focusout', 'pointerover', 'pointerout', 'pointerup', 'pointerdown'];
+	#passiveEvents = ['pointermove'];
+	#resizeCursor = 'ns-resize';
+	#pointerCursor = 'pointer';
+	#pointerCapture = false;
+	#ownerDocument = globalThis.document;
+	#canvasContext = globalThis.document?.createElement('canvas').getContext('2d');
+	#typeOfValue(value) {
+		if (/^([-+]?\d*\.?\d+)$/.test(value)) return 'number';
+		if (/^(%|cm|mm|in|px|pt|pc|em|ex|ch|rem|vw|vh|vmin|vmax|vb|vi|svw|svh|lvw|lvh|dvw|dvh|lh|rlh)$/.test(value)) return 'unit';
+		if (/^(\w+)$/.test(value)) return 'keyword';
+		if (/^(\w+\()/.test(value)) return 'function';
```

Comment:
haha just noticed this after commented above, still the question is if we want to parse CSS with regexes, the answer is pretty sure no, but I also understand the desire to keep it small, since we will need a CSS parser for for many places, we may as well start using some good dependency